storages/backends/gcloud.py

```diff
@@ -142,10 +144,18 @@ def get_default_settings(self):
             # roll over.
             "max_memory_size": setting("GS_MAX_MEMORY_SIZE", 0),
             "blob_chunk_size": setting("GS_BLOB_CHUNK_SIZE"),
+            "sa_email": setting("GS_SA_SIGNING_EMAIL")
         }
 
     @property
     def client(self):
+        credentials, project_id = auth.default(scopes=['https://www.googleapis.com/auth/cloud-platform'])
+        if self.project_id is None:
+            self.project_id = project_id
+        credentials.refresh(requests.Request())
+        if not hasattr(credentials, "service_account_email"):
```

Comment:
What determines if the `hasattr` check fails? Also, if it fails is it an error not to see `sa_email`?

Reply:
In some situations (workload identity) if you try to use the default service account the service account email can be missing. In this situation its best to provide the sa_email so you can sign urls.

Reply:
Sorry, you are saying that this *must* be set in some situation? Can you add that to the documentation.